docs/design/contextual-tabs.md

```diff
@@ -0,0 +1,392 @@
+
+A contextual tab is a hidden tab control in the Office ribbon that is displayed in the tab row when an object in the Office document, such as an image or a table has focus; for example, the **Table Design** tab that appears on the Excel ribbon when a table is selected. You can include custom contextual tabs in your Office add-in and specify when they are visible or hidden.
+
+> [!NOTE]
+> This article assumes that you are familiar with the following documentation. Please review it if you haven't worked with Add-in Commands (custom menu items and ribbon buttons) recently.
+>
+> - [Basic concepts for Add-in Commands](add-in-commands.md)
+
+> [!IMPORTANT]
+> Custom contextual tabs are in preview. Please experiment with them in a development or testing environment but don't add them to a production add-in.
+>
+> Custom contextual tabs are currently only supported on Excel and only on these platforms and builds:
+>
+> * Excel on Windows (Microsoft 365 only, not perpetual license): Version ???? (Build ?????.?????) Your Microsoft 365 subscription may need to be on the [Current Channel (Preview)](https://insider.office.com/en-us/join/windows) formerly called "Monthly Channel (Targeted)" or "Insider Slow".
+
+> [!NOTE]
+> Custom contextual tabs work only on platforms that support the following requirement sets. For more about requirement sets and how to work with them, see [Specify Office applications and API requirements](../develop/specify-office-hosts-and-api-requirements.md).
+>
+> - [SharedRuntime 1.1](../reference/requirement-sets/shared-runtime-requirement-sets.md)
+> - [RibbonApi 1.2](../reference/requirement-sets/ribbon-api-requirement-sets.md)
+
+## Behavior of custom contextual tabs
+
+The user experience for custom contextual tabs follows the pattern of built-in Office contextual tabs. The following are the basic principles for the placement custom contextual tabs:
+
+- When a custom contextual tab is visible, it appears on the right end of the ribbon.
+- If one or more built-in contextual tabs and one or more custom contextual tabs from add-ins are visible at the same time, the custom contextual tabs are always to the right of all of the built-in contextual tabs.
+- If your add-in has more than one contextual tab and there are contexts in which more than one is visible, they appear in the order (left-to-right) in which they are defined in your add-in. See [Define the groups and controls that appear on the tab](#define-the-groups-and-controls-that-appear-on-the-tab) for details about how you define them.
+- If more than one add-in has a contextual tab that is visible in a specific context, then they appear in the order (left-to-right) in which the add-ins were installed.
+
+## Major steps for including a contextual tab in an add-in
+
+The following are the major steps for including a custom contextual tab in an add-in:
+
+1. Configure the add-in to use a shared runtime.
+1. Define the tab and the groups and controls that appear on it.
```

Comment:
Define the tab(s), the group(s) and the control(s) with the following rules.
tab: 1 min to 10 max tab(s)
    group: 1 min to 6 max control(s) per group(s)
        control: 1 min to 20 max control per tab

example for a tab with 20 controls max following the rules above: 
   3 groups of 6 controls and 1 group of 2 controls
   5 groups of 4 controls 
   4 groups of 5 controls
   20 groups of 1 control